### PR TITLE
[16.0][MIG] edi_party_data_oca: Migration to 16.0

### PR DESCRIPTION
**Pull request by QuocDuong1306**
This was moved from:
- https://github.com/OCA/edi/pull/812

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+odoo_test_helper
+odoo-addon-edi_oca @ git+https://github.com/OCA/edi-framework.git@refs/pull/5/head#subdirectory=setup/edi_oca
```
Comment (simahawk) [inline test-requirements.txt]: you can drop this req

**Review by simahawk — COMMENTED** [review]
LG overall

**Comment by nilshamerlinck** [comment]
A few notes about `edi_party_data_oca` tests:
1) if we only run them alone, they pass
2) if we run tests for both `eci_oca` and `edi_party_data_oca`, they fail
3) in 14.0, they pass, but that's because [`@tagged("-at_install", "post_install")`](https://github.com/OCA/edi/blob/1c3a8d0172926078b346d175e4a19a575c56ee91/edi_oca/tests/common.py#L97) is not actually effective: tests run `at_install` (TODO)
4) in 16.0, `@tagged("-at_install", "post_install")` works so tests run on a fully loaded registry
 - then `FakeModelLoader` is used in 2 `edi_oca` tests: `test_consumer_mixin`, `test_security`
 - and it impacts the registry on `registry.load("edi_oca")` in both [`update_registry()`](https://github.com/OCA/odoo-test-helper/blob/9ccb57d7749605b9bb61e27f882c5959ef058b9f/odoo_test_helper/fake_model_loader.py#L142) and [`restore_registry()`](https://github.com/OCA/odoo-test-helper/blob/9ccb57d7749605b9bb61e27f882c5959ef058b9f/odoo_test_helper/fake_model_loader.py#L180): `env["edi.exchange.type"]` registry class is rebuilt ignoring [`edi_party_data_oca`'s extension](https://github.com/OCA/edi-framework/pull/20/files#diff-5bdc5cb39bc88b590c5c64e9cb5a631a961ac04cfab0c750179f0e78e133709cR12)
 - as a consequence, we get the error: `AttributeError: 'edi.exchange.type' object has no attribute 'id_category_ids'`
5) workarounds:
 - run `edi_party_data_oca` tests before `edi_oca` tests using `@tagged("at_install", "-post_install")`
 - force reload of `edi_party_data_oca` with FakeModelLoader (current chosen workaround)
6) long term fixes:
 - it seems to me that the second `registry.load("edi_oca")` in [`restore_registry()`](https://github.com/OCA/odoo-test-helper/blob/9ccb57d7749605b9bb61e27f882c5959ef058b9f/odoo_test_helper/fake_model_loader.py#L180) is not needed anymore in odoo >= 16.0, to be investigated further (inputs welcome!)

**Comment by simahawk** [comment]
@nilshamerlinck no time to look into details now... Can this help? https://github.com/OCA/odoo-test-helper/pull/27

**Comment by nilshamerlinck** [comment]
Hello @simahawk no, it's a different case; anyway, PR is green now with the workaround :) long term fixes can/will come later

**Comment by nilshamerlinck** [comment]
Hello @simahawk, opened https://github.com/OCA/odoo-test-helper/pull/28 to address the tests issues at the `odoo-test-helper` level instead of workarounds in impacted PRs

**Comment by etobella** [comment]
TEST Requirements are not needed, the branch has already been merged and deployed :smile:

**Comment by simahawk** [comment]
odoo-test-helper 2.1.1 released

File: test-requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 odoo_test_helper
+odoo-test-helper @ git+https://github.com/oca/odoo-test-helper@refs/pull/28/head
```
Comment (simahawk) [inline test-requirements.txt]: you can remove this as a new version has been released

**Comment by QuocDuong1306** [comment]
It's OK, thank @simahawk @etobella

**Review by simahawk — APPROVED** [review]

**Pull request merged**